corim/unsignedcorim.go

```diff
@@ -4,104 +4,219 @@
 package corim
 
 import (
+	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/veraison/corim/comid"
 	"github.com/veraison/eat"
 	"github.com/veraison/swid"
 )
 
+// UnsignedCorim is the top-level representation of the unsigned-corim-map with
+// CBOR and JSON serialization.
 type UnsignedCorim struct {
-	ID            ID             `cbor:"0,keyasint" json:"corim-id"`
+	ID            swid.TagID     `cbor:"0,keyasint" json:"corim-id"`
 	Tags          []Tag          `cbor:"1,keyasint" json:"tags"`
 	DependentRims *[]Locator     `cbor:"2,keyasint,omitempty" json:"dependent-rims,omitempty"`
 	Profiles      *[]eat.Profile `cbor:"3,keyasint,omitempty" json:"profiles,omitempty"`
 }
 
-type ID struct {
-	val interface{}
+// NewUnsignedCorim instantiates an empty UnsignedCorim
+func NewUnsignedCorim() *UnsignedCorim {
+	return &UnsignedCorim{}
 }
 
-func (o ID) MarshalCBOR() ([]byte, error) {
-	return em.Marshal(o.val)
+// SetID sets the corim-id in the unsigned-corim-map to the supplied value.  The
+// corim-id can be passed as UUID in string or binary form (i.e., byte array),
+// or as a (non-empty) string
+func (o *UnsignedCorim) SetID(v interface{}) *UnsignedCorim {
+	if o != nil {
+		tagID := swid.NewTagID(v)
+		if tagID == nil {
+			return nil
+		}
+		o.ID = *tagID
+	}
+	return o
 }
 
-func (o *ID) UnmarshalCBOR(data []byte) error {
-	var s string
+// GetID retrieves the corim-id from the unsigned-corim-map as a string
+func (o UnsignedCorim) GetID() string {
+	return o.ID.String()
+}
 
-	if dm.Unmarshal(data, &s) == nil {
-		o.val = s
-		return nil
-	}
+// AddComid appends the CBOR encoded (and appropriately tagged) CoMID to the
+// tags array of the unsigned-corim-map
+func (o *UnsignedCorim) AddComid(c comid.Comid) *UnsignedCorim {
+	if o != nil {
+		if c.Valid() != nil {
+			return nil
+		}
+
+		comidCBOR, err := c.ToCBOR()
+		if err != nil {
+			return nil
+		}
 
-	var u comid.UUID
+		// d9 01fa # tag(506)
+		comidTag := []byte{0xd9, 0x01, 0xfa}
```

Comment:
506 is 0x01, 0xFA, could you please also add the description for D9 on line 62?

Reply:
d9 is the `tag()` in the description

Reply:
Ah, I see!